### PR TITLE
Add serial-ymodem flash web-app (#77)

### PR DESCRIPTION
**Pull request by simonebortolin**

### CODE REVIEW

File: assets/js/rootLantiq.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ async function waitFailbackShell(writer, reader) {
                 writer.write('f\n');
             }, 10);
 
-            loading("Root in progress: Trigger characters received. Waiting for boot to end...",1);
+            outputMsgCallback("Root in progress: Trigger characters received. Waiting for boot to end...",1);
```
Comment (simonebortolin) [inline assets/js/rootLantiq.js:48]:
```suggestion
outputMsgCallback("Root in progress: Trigger characters received. Waiting for boot to end...");
```

**Comment by ErnyTech** [comment]
The changes to Huawei Webroot have been splitted into PR #97 to prevent this PR from getting too big

File: assets/js/rootLantiq.js [hunk 2]
```diff
@@ -131,3 +131,20 @@ async function unlockHuaweiShell(port, outputMsgCallback, outputErrorCallback, b
         return false;
     }
 }
+
+async function startYmodemLoad(port, outputMsgCallback, outputErrorCallback, baudRate = 115200, address = "0x80800000") {
```
Comment (simonebortolin) [inline assets/js/rootLantiq.js:135]: callbacks should be put at the end not in the middle

Comment (ErnyTech) [reply assets/js/rootLantiq.js:135]: We can't use the default parameters by moving the callbacks to the end

File: _ont/ont-huawei-ma5671a-ymodem.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    flashModal.addEventListener('modal-jtd-open', async function(event) {
+        flash({signal: cs});
+    });
+    function pause(message) {
+        flashTextStep.textContent = message;
+    }
+    function loading(message) {
+        flashTextStep.textContent = message;
+    }
+    function showError(message) {
```
Comment (simonebortolin) [inline _ont/ont-huawei-ma5671a-ymodem.md:68]: add the class error to flashTextStep

File: _ont/ont-huawei-ma5671a-ymodem.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+        flashTextStep.textContent = message;
+    }
+    function showSuccess(message) {
+        flashTextStep.textContent = message;
```
Comment (simonebortolin) [inline _ont/ont-huawei-ma5671a-ymodem.md:72]: add the class success to flashTextStep

File: assets/js/rootLantiq.js [hunk 5]
```diff
@@ -1,20 +1,28 @@
-async function waitUbootStop(writer, reader, sfpModel, outputMsgCallback) {
-    const interval = setInterval(function() {
-        writer.write(String.fromCharCode(3));
-    }, 10);
+const LOAD_ADDR = "80800000"
+const IMAGE0_ADDR = "C0000 740000";
```
Comment (simonebortolin) [inline assets/js/rootLantiq.js:2]: are these values the same for all sticks or only for lantiq MA/SF/FS?

File: assets/js/xymini.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+            await portWriter.write(block);
+        } else {
+            portWriter.write(new Uint8Array([EOF]));
+        }
+
+        const { value, done } = await portReader.read();
+
+        if (value[0] == ACK) {
+            if (!size) {
+                console.log("XYMini: End of transmission");
```
Comment (simonebortolin) [inline assets/js/xymini.js:119]: here is not to be called `progressCallback`?

File: assets/js/rootLantiq.js (hunk 5 above)
Comment (ErnyTech) [reply assets/js/rootLantiq.js:2]: Only for Lantiq/MA/SF/FS (rootLantiq.js only supports these)

File: assets/js/xymini.js (hunk 6 above)
Comment (ErnyTech) [reply assets/js/xymini.js:119]: No, here he has already finished sending data, when the receiver confirms the end of transmission the function sendXYMini returns

**Review by ErnyTech — APPROVED** [review]

**Pull request merged**